### ISSUE TITLE
Putting config files in designated folders

### ISSUE THREAD

**Issue by heyarne** [jooola/airsonic-cli#2]
There's the [XDG Base Directory Spec](http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html), which basically says local config files belong into `~/.config/$tool/...` or `/etc/$tool`, but I think we're talking about local config here. This way they're independent from the bin location.

Any thoughts on this? I think also maybe accepting environment vars for config might be a good idea.

**Comment by jooola** [comment]
> There's the XDG Base Directory Spec, which basically says local config files belong into ~/.config/$tool/... or /etc/$tool, but I think we're talking about local config here. This way they're independent from the bin location.

I agree that we not to use a standard configuration file path.But using `~./config/airsonic-cli/` seems weird to me. I would better suggest directly `~.airsonic-cli.yml`
Also this really depends on the use of the tools. The main problem is simply that the file contains full credentials to access the server API and I'm not comfortable to put it in the wrong place.

> I think also maybe accepting environment vars for config might be a good idea.
This will be my next enhancement, really not difficult to add !

**Comment by heyarne** [comment]
Why weird exactly? The spec tries to standardize by extending the unix filesystem spec, which I think is a good goal. It lowers risk of namespace collisions and there are quite a few tools that use it already (Fish Shell, GTK 3, Yarn, plus some GUI apps). Also you don't have an enormous list of files in your home directory. ;)

Btw, the spec also says that when folders don't exist, one should create them with permissions `0700`, which should solve the security concerns. I do understand the sensitivity of the tokens saved though.

**Comment by jooola** [comment]
`~./config/airsonic-cli/` seems more for complicated applications with multiple configuration files. I guess the best way will be put it @ `~.airsonic-cli.yml` with 600 permissions.
People can move it if they want !

All those thoughts just comes from my habits and theories and surely not from the conventions. So let's do what you suggested.

Or do you really think we should move it to the configuration folder ?

**Comment by heyarne** [comment]
I don't think this is urgent enough to make a decision anybody's uncomfortable with. I *do* think that it makes sense to follow standards and have a folder for all airsonic-related config (`~/.config/airsonic/...` or `/etc/airsonic`, idk), maybe once this becomes an official management tool (which I would love! I could just set up a cron job to scan my folders every 2nd wednesday 5 times when it's musicday or anything like that).

If you say adding env vars like AIRSONIC_USER or AIRSONIC_PASS is an easy fix maybe that's enough for now, no?

**Comment by jooola** [comment]
I think we better set up proper and solid base and them add functionalities rather than add functionalities and changing the base every 2 weeks. This will need some investigation and maybe multiple choices.

As matter of env vars I can do it right now it takes 3 minutes.
